src/datatype/index.js

```diff
@@ -203,13 +203,15 @@ export class DataType extends TypedEmitter {
   /**
    * Not yet implemented
    * @param {string | string[]} versionId
+   * @param {T} value
    */
-  async delete(versionId) {
+  async delete(versionId, value) {
```

Comment:
Shouldn't need to update the signature here (also this is prone to several issues around value integrity).  Internally, this method should be getting the existing doc via `getByVersionId()` and using that as the base value. Something along the lines of:

```ts
  async delete(versionId) {
    await this.#dataStore.indexer.idle()
    const links = Array.isArray(versionId) ? versionId : [versionId]
    const { docId, createdAt, createdBy } = await this.#validateLinks(links)

    const existingDoc = await this.getByVersionId(links[links.length - 1])

    /** @type {any} */
    const doc = {
      ...existingDoc,
      docId,
      createdAt,
      updatedAt: new Date().toISOString(),
      createdBy,
      links,
      schemaName: this.#schemaName,
      deleted: true,
    }
    await this.#dataStore.write(doc)
    return this.getByDocId(docId)
  }
```

technically could update `#validateLinks` to return the existing doc in its entirety to avoid doing repeat work, but not sure if i'd suggest that given that it mixes concerns a bit

Reply:
yeah getting the doc by `versionId` makes sense (I was just passing the value to match `dataType.update`).